test/unit.js

```diff
@@ -8,8 +14,55 @@ describe('skatejs-web-components', () => {
         this.attachShadow({ mode: 'open' });
       }
     };
-    window.customElements.define('x-test', Elem);
+    customElements.define('x-test-1', Elem);
     const elem = new Elem();
     expect(!!elem.shadowRoot).to.equal(true);
   });
+
+  patch && describe('#16 - safari recalc bug', (done) => {
+    function getDescriptor(name) {
+      return Object.getOwnPropertyDescriptor(Element.prototype, name) ||
+        Object.getOwnPropertyDescriptor(HTMLElement.prototype, name);
+    }
+
+    it('configurable', () => {
+      expect(getDescriptor('attachShadow').configurable).to.equal(true);
+    });
+
+    it('enumerable', () => {
+      expect(getDescriptor('attachShadow').enumerable).to.equal(true);
+    });
+
+    it('writable', () => {
+      expect(getDescriptor('attachShadow').writable).to.equal(true);
+    });
+
+    it('functionality', () => {
+      const div = document.createElement('div');
+      div.attachShadow({ mode: 'open' });
+
+      const style = document.createElement('style');
+      div.shadowRoot.appendChild(style);
+
+      const p = document.createElement('p');
+      div.shadowRoot.appendChild(p);
+      p.appendChild(document.createTextNode('testing'));
+
+
+      body.appendChild(div);
+      // Wait for the first flow / layout the style element will cause.
+      setTimeout(() => {
+        // Then set the style content.
+        style.textContent = 'p { border: 1px solid rgb(0, 0, 0); }';
+
+        // We must wait until the next reflow / relayout updating the style element causes to then
+        // check if it appolied.
+        setTimeout(() => {
+          expect(window.getComputedStyle(p).border).to.equal('1px solid rgb(0, 0, 0)');
```

Comment:
Can you maybe check this for the second time? We had quite a few issues when something was working just once when it supposed to work all the time.
Also the empty case  `style.textContent = ''` - that all the styles are removed properly.